### PR TITLE
cdc: fix deregister ABA problem and support cancel incremental scan 

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Problem Summary:

1. cdc: deregister region only when ObserveID matches
2. cdc: add a metric that record captured region number 
3. cdc: support cancel incremental scan

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

Fix deregister ABA problem and support cancel incremental scan.

### CODE REVIEW

File: components/cdc/src/endpoint.rs [hunk 1]
```diff
@@ -584,6 +652,13 @@ impl Initializer {
             .unwrap();
         let mut done = false;
         while !done {
+            if !self.cancel.load(Ordering::SeqCst) {
```
Comment (5kbpers) [inline components/cdc/src/endpoint.rs:655]: As I understand, it should be named `continue`

**Comment by overvenus** [comment]
PTAL

File: components/cdc/src/endpoint.rs [hunk 2]
```diff
@@ -312,12 +359,13 @@ impl<T: 'static + RaftStoreRouter> Endpoint<T> {
             return;
```
Comment (MyonKeminta) [inline components/cdc/src/endpoint.rs:359]: Could you add a log here by the way?

**Pull request merged**